Assignment/Assignment.Tests/NodeTests.cs

```diff
@@ -0,0 +1,111 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Assignment
+{
+    [TestClass]
+    public class NodeTests
+    {
+        [TestMethod]
+        public void TestGetEnumerator()
+        {
+            // Arrange
+            Node<int> node = new Node<int>(1);
+            node.Add(2);
+            node.Add(3);
+
+            // Act
+            var result = node.GetEnumerator();
+            result.MoveNext();
+            int firstValue = result.Current;
+            result.MoveNext();
+            int secondValue = result.Current;
+            result.MoveNext();
+            int thirdValue = result.Current;
+
+            // Assert
+            Assert.AreEqual(1, firstValue);
```

Comment:
Don't forget to use the generic version, like this:
```suggestion
            Assert.AreEqual<int>(1, firstValue);
```